### PR TITLE
BaSP-MR-61: subscriptions add and divide components

### PR DESCRIPTION
**Pull request by Dannylez**
- New file structure using index and form
- Incroporation of shared components
- Use of React Router
- Styles for de new implementations
- Loading message while data is being received

### CODE REVIEW

**Review by DelfiLT — CHANGES_REQUESTED** [review]
When you try to edit or add a suscription the app crashes.
For UI corrections you should add to the background image   `background-repeat: no-repeat;`
 ` background-position: center;`

**Comment by ukiweskamp** [comment]
> When you try to edit or add a suscription the app crashes. For UI corrections you should add to the background image `background-repeat: no-repeat;` ` background-position: center;`

I haven't had a chance to test the code yet but Danny made changes to the server and did a merge. That's probably why it won't work for you.

File: src/Components/Layout/index.jsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ import Admins from '../Admins';
 import Classes from '../Classes';
 import Members from '../Members';
 import Subscriptions from '../Subscriptions';
+import Form from '../Subscriptions/Form';
```
Comment (Bigzufi) [inline src/Components/Layout/index.jsx:9]: You shouldn't use such a generic name like Form, try using SubscriptionForm or SubsForm.

**Comment by Dannylez** [comment]
> When you try to edit or add a suscription the app crashes. For UI corrections you should add to the background image `background-repeat: no-repeat;` ` background-position: center;`

Fixed, it doesn't crash anymore, and for the background I've used "background-size: cover" now, I hope you like it

**Review by DelfiLT — APPROVED** [review]

**Review by SantiH97 — APPROVED** [review]
Overall the code seems pretty well implemented, its clean, I didn't find anything unnecessary or too hard to read. I will leave two suggestions regarding presentation:
1- I don't think you should be showing the deleted item's id on deletion as it provides no usefull information for the end user
2 - I there's no data to show, either because it isnt loading or because the isn't any, you get the same message, so theoretically you can be waiting for a very long time without ever knowing if the data is going to show or there simply isnt any.

**Comment by Dannylez** [comment]
> Overall the code seems pretty well implemented, its clean, I didn't find anything unnecessary or too hard to read. I will leave two suggestions regarding presentation:
> 1- I don't think you should be showing the deleted item's id on deletion as it provides no usefull information for the end user
> 2 - I there's no data to show, either because it isnt loading or because the isn't any, you get the same message, so theoretically you can be waiting for a very long time without ever knowing if the data is going to show or there simply isnt any.

Ok, about the first point: I think it's something to fix from the server what include on the message when a deletion is made.
About the second point: I think you're right, maybe I can add a new state that returns true when the Get request is completed and customize the message either is still waiting a response or not

**Review by ukiweskamp — APPROVED** [review]

**Comment by Dannylez** [comment]
[Subscription - Add and Divide Components](https://trello.com/c/6dOOaFmB/61-subscription-add-and-divide-components)

**Pull request merged**